### ISSUE TITLE
Subscription created but no files added to wmbs tables

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#8751]
and thus, no jobgroup associated.

TO BE FOLLOWED UP AND FIXED: how to handle blocks where all files are marked as invalid.

This workflow
prebello_Run2018A-v1-ZeroBias-16Aug2018_1020_180816_184255_5826

is stuck in vocms0257 because JobCreator hasn't created any jobs for subscription 269, which is related to this GWQE:
https://cmsweb.cern.ch/couchdb/workqueue/_design/WorkQueue/_rewrite/element/5d7ac0ee01bceb88b62380baf7f47e5e

That block has these runs (from DBS):
https://cmsweb.cern.ch/dbs/prod/global/DBSReader/runs?block_name=/ZeroBias/Run2018A-v1/RAW%2300009dbf-ac2a-4da3-b807-b3160c051bce

and the run number intersection with the RunWhitelist is:
```
In [11]: list(set(reqmgr).intersection(dbs))
Out[11]: [316569, 316613, 316590, 316615]
```

Picking the first run, you can see there are around 10 files that could have been acquired:
https://cmsweb.cern.ch/dbs/prod/global/DBSReader/files?block_name=/ZeroBias/Run2018A-v1/RAW%2300009dbf-ac2a-4da3-b807-b3160c051bce&run_num=316569

WAIT!!! all those files are invalid, checking again for valid files only:
https://cmsweb.cern.ch/dbs/prod/global/DBSReader/files?block_name=/ZeroBias/Run2018A-v1/RAW%2300009dbf-ac2a-4da3-b807-b3160c051bce&run_num=316569&validFileOnly=1

Bottom line is, there is nothing to process in this block, but still the agent is not completing this subscription. We need to find a way to properly finish this subscription, for two reasons:
1) it will make WorkQueueManager skip WMBS data acquisition, since the subscription does not have any jobgroup associated with
2) workflow will never complete (actually, I think TaskArchiver will probably mark it as finished when time comes).

**Comment by amaltaro** [comment]
@ticoann I'm wondering whether we should just mark that subscription as finished? Do you see any possible side effects?

FYI @mcremone @Thongonary

**Comment by ticoann** [comment]
Nice findings Alan, I wouldn't expect that there will be all the files invalid in given block. I don't think any side effect on top of my head if  you mark that subscription finished. But I will go through the code to check in detail. The other concern is what would we do to prevent this happening again.

**Comment by amaltaro** [comment]
About preventing it from happening, maybe we can check whether there are any wmbs files to be inserted, somewhere near this code:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/WMBSHelper.py#L458

and if not, at the end of the wmbsPreparation we also mark the subscription as finished.

Second option would be to catch it upstream (GQ) and not event create a GQE, but then there is always a chance for a race condition between GQ and WMBS data acquisition.

**Comment by ticoann** [comment]
Actually, I am not which option is better. For the second option I don't think there will be race condition when workqueue element is created it also calculate the estimate job if job estimate is 0 due to no files in block we don't create the element or set as done or something. 

Unless, files are originally valid and change to invalid there won't be race condition. (I might be wrong).
I would choose the option which is has less changes (whichever that is)

**Comment by amaltaro** [comment]
I took a look at the code and it seems to be safe to mark that subscription as finished, so I just did so
`UPDATE wmbs_subscription SET finished = 1, last_update = 1535031925 WHERE id = 269;`

and now I see plenty of work getting acquired into wmbs.

I'm updating the first description of this issue with the generic problem that we still have to fix, though no longer high priority imo.

**Comment by ticoann** [comment]
It seems actually we check this status and skip those blocks in Global Queue,
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/Policy/Start/Block.py#L123
So files must be invalid after we pull the workflow GQ.

**Comment by ticoann** [comment]
I am closing this since there will be always the case if it making the file invalid while workflow is running on that input.
